src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -366,9 +373,6 @@ public boolean apply(@Nullable Container container) {
      *
      */
     private synchronized boolean addProvisionedSlave(String ami, int amiCap) throws Exception {
-        if( amiCap == 0 )
-            return true;
-
```

Comment:
For templates i have simple patch that allows set the same label for multiple amis.
Also, you can't define two instances with the same ami - either fetch of the number running instances wouldn't work.